### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lukacsja**
[DEMO LINK](https://lukacsja.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -35,61 +36,134 @@ export function getReorderedGoods(
   const visibleGoods = [...goods];
 
   // Sort and reverse goods if needed
```
Comment (kenvastolord) [inline src/App.tsx:38]: Good job solving this task! Everything seems to be excellent, just one little thing, do not forget to remove the comments, it  is not important but make sure to delete it before push the code.

File: src/App.tsx [hunk 2]
```diff
@@ -35,61 +36,134 @@ export function getReorderedGoods(
...
+              'button is-warning',
+              {
+                'is-light': !isReversed,
+              },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+          {(isReversed || sortType !== SortType.NONE) && (
```
Comment (kenvastolord) [inline src/App.tsx:147]: it would be better if you save ( isReversed || sortType !== SortType.NONE) in a variable and use it here.

Comment (lukacsja) [reply src/App.tsx]: Actually i liked the comments. It made the whole code so much readable for me but you are right at least i should have rewritten some of them.

I implemented your other idea as well.

File: src/App.tsx [hunk 3]
```diff
@@ -26,70 +27,140 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
```
Comment (ysrckr) [inline src/App.tsx:33]: If you make a type for this function, it would be more readable.

File: src/App.tsx [hunk 4]
```diff
@@ -26,70 +27,140 @@ type ReorderOptions = {
...
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
```
Comment (ysrckr) [inline src/App.tsx:56]: Let's remove console statement in the final code.

**Review by ysrckr — CHANGES_REQUESTED** [review]
Other than those 2 thing looks good to me

File: src/App.tsx [hunk 5]
```diff
@@ -26,70 +27,137 @@ type ReorderOptions = {
...
+              'button is-warning',
+              {
+                'is-light': !isReversed,
+              },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+          {(isResetDisplayed) && (
```
Comment (naziamoff) [inline src/App.tsx:141]:
```suggestion
          {isResetDisplayed && (
```

File: src/App.tsx [hunk 6]
```diff
@@ -26,70 +27,137 @@ type ReorderOptions = {
...
+        </div>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {
+            visibleGoods.map(good => (
```
Comment (naziamoff) [inline src/App.tsx:153]:
```suggestion
          {visibleGoods.map(good => (
```

**Review by naziamoff — APPROVED** [review]
well done!
@kenvastolord @ysrckr good job with reviewing 👍

**Pull request closed without merging**